### ISSUE TITLE
Expose enums in documentation in a way that allows hyperlinks

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#13254]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Any OS, current master branch (f5c513ca7)

**Issue description:**
We have support for keeping track of enums in the classref's XML, with this form:
```
Light2D.xml
28:             <member name="mode" type="int" setter="set_mode" getter="get_mode" enum="Light2D.Mode">
61:             <member name="shadow_filter" type="int" setter="set_shadow_filter" getter="get_shadow_filter" enum="Light2D.ShadowFilter">
```

But currently there's no real link done with the actual constants from that enum:
```
Light2D.xml
29:                     The Light2D's mode. See MODE_* constants for values.
81:             <constant name="MODE_ADD" value="0">
84:             <constant name="MODE_SUB" value="1">
87:             <constant name="MODE_MIX" value="2">
90:             <constant name="MODE_MASK" value="3">
```

We should add a way to properly show and reference enums in the oflfine and online docs, so that doc writers can write `[enum Light2D.Mode]` to create a reference to the enum section where the relevant constants are defined.

**Comment by akien-mga** [comment]
Actually the enums are properly displayed already in the editor, though the references are missing in the XML tags:
![spectacle fb4852](https://user-images.githubusercontent.com/4701338/33222601-c0a44ede-d158-11e7-9b01-f7e02503b98a.png)

For the online docs, we still need to implement it: http://docs.godotengine.org/en/latest/classes/class_light2d.html#numeric-constants

**Comment by akien-mga** [comment]
> Actually the enums are properly displayed already in the editor, though the references are missing in the XML tags

That was a bug, now fixed with 2bc6db65 and 7dfba3cd.

So now we just need to replicate what the engine does with this metadata in `makerst.py`.
